internal/otelmatchers/trace_matchers_test.go

```diff
@@ -0,0 +1,240 @@
+package otelmatchers
+
+import (
+	"os"
+	"testing"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func TestCustomMatcher(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Custom Matcher Suite")
+}
+
+var _ = Describe("ConsistOfSpansWithIDs", func() {
+	var fileBytes []byte
+	var expectedSpansWithIDs []pcommon.SpanID
+
+	BeforeEach(func() {
+		expectedSpansWithIDs = []pcommon.SpanID{[8]byte{1}, [8]byte{2}, [8]byte{3}}
+	})
+
+	Context("with nil input", func() {
+		BeforeEach(func() {
+			fileBytes = nil
+		})
+
+		It("should fail", func() {
+			Expect(fileBytes).ShouldNot(ConsistOfSpansWithIDs(expectedSpansWithIDs))
+		})
+	})
+
+	Context("with input of invalid type", func() {
+		It("should error", func() {
+			_, err := ConsistOfSpansWithIDs(expectedSpansWithIDs).Match(struct{}{})
+			Expect(err).Should(HaveOccurred())
+		})
+	})
+
+	Context("with no spans matching the span IDs", func() {
+		BeforeEach(func() {
+			var err error
+			fileBytes, err = os.ReadFile("testdata/consist_of_spans_with_ids/no_match.jsonl")
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should fail", func() {
+			Expect(fileBytes).ShouldNot(ConsistOfSpansWithIDs(expectedSpansWithIDs))
+		})
+	})
+
+	Context("with some spans matching the span IDs", func() {
+		BeforeEach(func() {
+			var err error
+			fileBytes, err = os.ReadFile("testdata/consist_of_spans_with_ids/partial_match.jsonl")
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should fail", func() {
+			Expect(fileBytes).ShouldNot(ConsistOfSpansWithIDs(expectedSpansWithIDs))
+		})
+	})
+
+	Context("with all spans matching the span IDs", func() {
+		BeforeEach(func() {
+			var err error
+			fileBytes, err = os.ReadFile("testdata/consist_of_spans_with_ids/full_match.jsonl")
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should succeed", func() {
+			Expect(fileBytes).Should(ConsistOfSpansWithIDs(expectedSpansWithIDs))
+		})
+	})
+
+	Context("with invalid input", func() {
+		BeforeEach(func() {
+			fileBytes = []byte{1, 2, 3}
+		})
+
+		It("should error", func() {
+			_, err := ConsistOfSpansWithIDs(expectedSpansWithIDs).Match(fileBytes)
```

Comment:
To fully test the contract of the Function Under Test (`ConsistOfSpansWithIDs`), I suggest to ensure that the returned list of span IDs is nil is error has occured:

```go
spans, err := ConsistOfSpansWithIDs(expectedSpansWithIDs).Match(fileBytes)

Expect(err).Should(HaveOccurred())
Expect(spans).ToBeNil()
```

Reply:
The `Match` method does not return spans, it always returns a `success` flag, which is then internally checked by Gomega (you would never call `Match` explicitly). I added a check for the flag.